src/superpmi-collect/superpmicollect.cs

```diff
@@ -0,0 +1,777 @@
+
+                // The way tests are run on Linux, we might need to do some setup. In particular,
+                // if the test scripts are copied from Windows, we need to convert line endings
+                // to Unix line endings, and make the script executable. We can always do this
+                // more than once. This same transformation is done in runtest.sh.
+                // Review: RunProgram doesn't seem to work if the program isn't a full path.
+
+                RunProgram("/usr/bin/perl", @"-pi -e 's/\r\n|\n|\r/\n/g' " + "\"" + testName + "\"");
+                RunProgram("/bin/chmod", "+x \"" + testName + "\"");
+
+                // Now, figure out how to run the test.
+
+                int lastIndex = testName.LastIndexOf("/");
+                if (lastIndex == -1)
+                {
+                    throw new SpmiException("test path doesn't have any directory separators? " + testName);
+                }
+                testDir = testName.Substring(0, lastIndex);
+            }
+
+            // Run the script in the same directory where the test lives.
+            string originalDir = Directory.GetCurrentDirectory();
+            Directory.SetCurrentDirectory(testDir);
+
+            try
+            {
+                RunProgram(testName, "");
+            }
+            finally
+            {
+                // Restore the original current directory from before the test run.
+                Directory.SetCurrentDirectory(originalDir);
+            }
+        }
+
+        // Run all the programs from the CoreCLR test binary drop we wish to run while collecting MC files.
+        private static void RunTestProgramsWhileCollecting()
+        {
+            // The list of all programs from the CoreCLR repo test binary drop that
+            // we should run when doing the SuperPMI collection. This is currently a
+            // hard-coded list of the relative paths within the test build binaries
+            // directory of the Windows .cmd files used to run a test. For non-Windows
+            // platforms, the .cmd is replaced by .sh, and the path separator character
+            // is changed.
+            //
+            // TODO: this should probably be loaded dynamically from a .json/.xml file.
+            string[] SuperPMICollectionTestProgramsList =
+            {
+                @"JIT\Performance\CodeQuality\Roslyn\CscBench\CscBench.cmd",
+                @"JIT\Methodical\fp\exgen\10w5d_cs_do\10w5d_cs_do.cmd",
+                @"JIT\Generics\Coverage\chaos65204782cs\chaos65204782cs.cmd"
+            };
+
+            // Figure out the root of the test binaries directory.
+            // Perhaps this (or something similar) would be a better way to figure out the binary root dir:
+            // testBinaryRootDir = System.IO.Path.GetDirectoryName(new Uri(Assembly.GetExecutingAssembly().CodeBase).LocalPath);
+
+            string thisTestDir = Directory.GetCurrentDirectory();
```

Comment:
I'm going to assume at some point we change dirs to a test directory before running this function?